src/resolve/resolved-dns-scope.c

```diff
@@ -1428,6 +1428,14 @@ int dns_scope_announce(DnsScope *scope, bool goodbye) {
         if (scope->protocol != DNS_PROTOCOL_MDNS)
                 return 0;
 
+        r = sd_event_get_state(scope->manager->event);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to get event loop state: %m");
+
+        /* If this is called on exit, through manager_free() -> link_free(), then we cannot announce. */
+        if (r == SD_EVENT_FINISHED)
+                return 0;
```

Comment:
With this patch applied resolved no longer sends goodbye packets when it's stopped. services announced via resolved get stuck in caches until they either auto-expire or are reconfirmed after failed attempts to connect to those services.

I think this PR should be reverted.

Reply:
Filed as #30421. I will take a look.

Reply:
Given that it isn't the first time PRs have gotten merged with no meaningful review I think it would make sense to enforce some sort of policy preventing those YOLO reviews followed by those YOLO merges from happening. One option would be to use CODE-OWNERS. I have no idea why PRs affecting networkd and resolved are "reviewed" with basically no knowledge of what goodbye packets or router advertisements are.

Reply:
In this case, I forget to paste your comment in another PR. Sorry.
But, at least, with this change, the state is better than previous, as this avoids to trigger assert_return(), even if it is not critical at the time.

Anyway, let's discuss at #30421 further fix or improvement.

Reply:
>  paste your comment in another PR

I don't think it would have helped because those sloppy merges happen even with full backtraces, pcaps and so on. They can't be stopped by adding more comments. The only thing that can stop them is the CI (because the usual "review" involves just pressing the button if the CI is green) so I guess before fixing that issue it would probably make sense to revamp the mDNS testsuite (and as far as I can remember there is a PR somewhere attempting to do exactly that. plus it comes with features that should make it easier to keep track of goodbye packets among other things).

> let's discuss at https://github.com/systemd/systemd/issues/30421 further fix or improvement

Let's get past https://github.com/systemd/systemd/pull/30049 first. It should help to prevent sloppy merges too.